sequencer/src/l1_client.rs

```diff
@@ -276,140 +168,50 @@ async fn get_finalized_block<P: JsonRpcClient>(
-                    "snapshotted finalized {} does not match expected finalized {}",
-                    snapshot_finalized.number,
-                    expected_finalized_number
-                );
-                if expected_finalized_number == snapshot_finalized.number {
-                    assert_eq!(expected_finalized.hash.unwrap(), snapshot_finalized.hash);
-                    assert_eq!(expected_finalized.timestamp, snapshot_finalized.timestamp);
-                    tracing::info!("L1 finalized block fully synchronized");
-                }
-            } else {
-                tracing::info!("no finalized L1 block yet");
-                assert_eq!(snapshot.finalized, None);
-            }
-
-            // Check invariants: monitonically increasing L1 blocks.
-            assert!(snapshot.head >= prev_head);
-            if let Some(finalized) = snapshot.finalized {
-                if let Some(prev_finalized) = prev_finalized {
-                    assert!(finalized.number >= prev_finalized.number);
-                }
-            } else {
-                assert_eq!(prev_finalized, None);
-            }
-
-            prev_head = snapshot.head;
-            prev_finalized = snapshot.finalized;
-
-            sleep(test_interval).await;
-        }
-    }
+    use super::*;
+    use crate::NodeState;
+    use ethers::utils::Anvil;
 
     #[async_std::test]
-    async fn test_reorg_increasing_block() {
-        setup_logging();
-        setup_backtrace();
+    async fn test_l1_block_fetching() -> anyhow::Result<()> {
+        // Test l1_client methods against `ethers::Provider`. There is
+        // also some sanity testing demonstrating `Anvil` availability.
+        let anvil = Anvil::new().block_time(1u32).spawn();
+        let l1_client = L1Client::new(anvil.endpoint().parse().unwrap());
+        let provider = &l1_client.provider;
+
+        let version = provider.client_version().await.unwrap();
+        assert_eq!("anvil/v0.2.0", version);
+
+        // Test that nothing funky is happening to the provider when
+        // passed along in state.
+        let state = NodeState {
+            l1_client: L1Client::new(anvil.endpoint().parse().unwrap()),
+            ..Default::default()
+        };
+        let version = state.l1_client().provider.client_version().await.unwrap();
+        assert_eq!("anvil/v0.2.0", version);
 
-        let mut anvil = AnvilOptions::default()
-            .block_time(Duration::from_secs(1))
-            .spawn()
-            .await;
-        let l1_client = L1ClientOptions::with_url(anvil.ws_url())
-            .start()
-            .await
-            .unwrap();
+        // compare response of underlying provider w/ `get_block_number`
+        let expected_head = provider.get_block_number().await.unwrap().as_u64();
+        let head = l1_client.get_block_number().await;
+        assert_eq!(expected_head, head);
```

Comment:
Technically these can return different numbers if the L1 block changes in between the two calls. Maybe this is so unlikely that we don't care. Not sure.

Reply:
Yes I thought of that. Its the kind of thing that may cause occasional failures requiring us to rerun the tests to get them to pass. The window is 1 sec wide, so I don't *think* it will happen very frequently. If these methods had fail cases we could just `unwrap` and remove the assert statements. As they are, I'm not thinking of a better way to test them.